blosc/blosc2.c

```diff
@@ -2315,7 +2323,7 @@ int blosc_compress_context(blosc2_context* context) {
   if (context->blosc2_flags & BLOSC2_INSTR_CODEC) {
     int dont_split = (context->header_flags & 0x10) >> 4;
     int blocksize = dont_split ? sizeof(blosc2_instr) : sizeof(blosc2_instr) * context->typesize;
-    _sw32(context->dest + BLOSC2_CHUNK_NBYTES, nstreams * sizeof(blosc2_instr));
+    _sw32(context->dest + BLOSC2_CHUNK_NBYTES, context->nblocks * blocksize);
```

Comment:
We want to use nstreams, so use the original line here.

Reply:
solved at f2fdffa71814dd1cbda6d3388e551c7e61214f64